tsdb/engine/v1/engine_test.go

```diff
@@ -0,0 +1,113 @@
+package v1_test
+
+import (
+	"io/ioutil"
+	"os"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/influxdb/influxdb/tsdb"
+	"github.com/influxdb/influxdb/tsdb/engine/v1"
+)
+
+// Ensure the engine can write duplicate points to the WAL and cache and retrieve them correctly.
+func TestEngine_Cursor_Duplicate(t *testing.T) {
+	e := OpenDefaultEngine()
+	defer e.Close()
+
+	// Write point.
+	if err := e.WritePoints([]tsdb.Point{
+		tsdb.NewPoint("cpu", tsdb.Tags{}, tsdb.Fields{"value": 100}, time.Unix(0, 1)),
+	}, nil, nil); err != nil {
+		t.Fatal(err)
+	}
+
+	// Flush to disk.
+	if err := e.Flush(0); err != nil {
+		t.Fatal(err)
+	}
+
+	// Write point again.
+	if err := e.WritePoints([]tsdb.Point{
+		tsdb.NewPoint("cpu", tsdb.Tags{}, tsdb.Fields{"value": 100}, time.Unix(0, 1)),
```

Comment:
second point should have a different value, which you can then read in the test verification to ensure that the latter point is the one that was actually used when a dupe was found